casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala

```diff
@@ -51,7 +51,7 @@ object BlockAPI {
 
   def deploy[F[_]: Concurrent: EngineCell: Log: Span](
       d: Signed[DeployData],
-      triggerPropose: Option[Casper[F] => F[Option[Int]]]
+      triggerPropose: Option[Casper[F] => F[Option[BlockHash]]]
```

Comment:
Is this commit relevant?

Reply:
Yes. What is the problem?

Reply:
It is outdated. I am curious how this comes.

Reply:
shouldn't  https://github.com/rchain/rchain/pull/3352/commits/6dc22c81a8cb31f996583bd550158bf8dd3417d7  be squashed to https://github.com/rchain/rchain/pull/3352/commits/5fca71c44b71395a7c240ac9deb84dd0723b7d7a ?they are changing the same parts of the code, and first commit is basically overwritten? Or I miss smth?

Reply:
I don't understand why you look at the outdated commit. I don't understand how you find the https://github.com/rchain/rchain/commit/6dc22c81a8cb31f996583bd550158bf8dd3417d7 here. Did you review before I assign you?

I suggest you just review whatever it is right now.

Reply:
I'm not sure - are yo saying its relevant or not? Both commits are in this PR, and second one introduces async and sync versions. Why to have first commit that reverts current async to sync? Why not introduces async and sync versions right away but first revert to sync? It touches the same parts of the code, right?

Reply:
Ok. I am going to squash it. We are playing words game.

Reply:
I'm not sure its words game - I simply do not understand whats going on :) If one commit is changing a line, then another commit is changing the same line and nothing else - I believe its reasonable to not have these two separate. There should be only the last commit.

Reply:
> I'm not sure its words game - I simply do not understand whats going on :) If one commit is changing a line, then another commit is changing the same line and nothing else - I believe its reasonable to not have these two separate. There should be only the last commit.

It doesn't make sense to me, though.